### PR TITLE
[API Compatibility] 第八批 API 对齐

### PR DESCRIPTION
**Pull request by zhwesky20**
### 第一批
- torch.nn.functional.instance_norm
- torch.nn.Module.to_empty
- torch.linalg.det
- torch.nn.functional.gumbel_softmax
- torch.vstack
- torch.Tensor.addcdiv
- torch.nn.functional.batch_norm
- torch.set_default_device
- torch.autograd.grad_mode.set_grad_enabled
- torch.Tensor.new_tensor
- torch.cuda.amp.GradScaler
- torch.nn.modules.loss._Loss
- torch.nn.modules.utils._pair

### 第二批
- torch.histc
- torch.masked_fill
- torch.hstack
- torch.Tensor.true_divide_
- torch.linalg.eigh
- torch.Tensor.H
- torch.nn.ELU
- torch.nn.functional.relu6
- torch.clamp_max
- torch.qr
- torch.logdet
- torch.linalg.cholesky
- torch.linalg.cross

### 第三批
- torch.nn.functional.elu
- torch.tensordot
- torch.real
- torch.nn.functional.prelu
- torch.nn.functional.rms_norm
- torch.expand_copy
- torch.Tensor.expm1_
- torch.moveaxis
- torch.linalg.qr
- torch.clamp_

### CODE REVIEW

File: python/paddle/compat/nn/functional/__init__.py [hunk 1]
```diff
@@ -470,3 +472,124 @@ def smooth_l1_loss(
...
+    """
+    return paddle.nn.functional.instance_norm(
+        x=input,
+        running_mean=running_mean,
+        running_var=running_var,
+        weight=weight,
+        bias=bias,
+        use_input_stats=use_input_stats,
+        momentum=1.0 - momentum,
+        eps=eps,
```
Comment (risemeup1111) [inline python/paddle/compat/nn/functional/__init__.py:594]: P1 **优先级：P1**

这里把 PyTorch 兼容签名里的 `running_mean` / `running_var` / `use_input_stats` 原样转给 `paddle.nn.functional.instance_norm`，但底层 Paddle 实现并不会使用这些参数：动态图路径只调用 `_C_ops.instance_norm(x, weight, bias, eps)`，静态图也只传 `X` / `Scale` / `Bias` 和 `epsilon` / `momentum` / `data_format`。这样用户调用 `instance_norm(..., running_mean=..., running_var=..., use_input_stats=False)` 时，PyTorch 会按 running stats 做推理归一化，而这里仍按当前输入统计量计算，结果会静默错误。

请实现 `use_input_stats=False` 的 running-stat 分支（并补充覆盖该路径的测试），或者在暂不支持时对 `use_input_stats=False` / 非空 running stats 显式报错，避免伪兼容。实现形态可以类似：

```python
if not use_input_stats:
    if running_mean is None or running_var is None:
        raise ValueError("running_mean and running_var are required when use_input_stats=False")
    stat_shape = [1, input.shape[1]] + [1] * (input.ndim - 2)
    mean = running_mean.reshape(stat_shape)
    var = running_var.reshape(stat_shape)
    out = (input - mean) / paddle.sqrt(var + eps)
    if weight is not None:
        out = out * weight.reshape(stat_shape)
    if bias is not None:
        out = out + bias.reshape(stat_shape)
    return out
```

**Review by risemeup1111 — CHANGES_REQUESTED** [review]
发现一个需要修复的兼容性问题，细节已放在行内评论中。核心风险是新增的 `instance_norm` 兼容 wrapper 接受了 PyTorch running-stat 推理参数，但当前实现会静默走输入统计量路径。

<div align="right">
   <sup>Powered by Nyanpasu with gpt-5.5 xhigh, please check the suggestions carefully.</sup>
</div>
